src/mon/AuthMonitor.cc

```diff
@@ -803,6 +807,20 @@ bool AuthMonitor::preprocess_command(MonOpRequestRef op)
   cmd_getval(g_ceph_context, cmdmap, "format", format, string("plain"));
   boost::scoped_ptr<Formatter> f(Formatter::create(format));
 
+
+  list<AuthMonReadCommandRef> read_cmds = {
```

Comment:
It may be worthwhile to consider that this will start becoming more complicated as types are added, and in this case it may be worth seeing if we can reduce some of the verbosity. The blanket advice to avoid using operator new with std::shared_ptr<> isn't bad in general (although it's more complicated than
it looks-- and after C++17 it almost just depends on what aspect you want to focus on-- I like to avoid bare operator new almost whenever possible, but
the penality of the extra allocation that used to exist is no longer the concern), but here I think we can focus on considering how much code is needed
to add a command. 

But what I'm getting at is that using make_shared<> here, it starts to get simpler. Further, since there's an emergent pattern, we can can use a 
tiny bit of metaprogramming and make it very straightforward and concise for the majority of cases-- with the additional advantage that over time 
if we need to add or remove a parameter for most command types, there's no need to modify all the calls in the list. 

Anyway, they all work and make different exchanges. Here's a comparison sandbox to see exactly what I mean:

https://godbolt.org/z/Xo56BR

Reply:
Also please consider std::vector for collections of pointers rather than std::list, unless you have a strong need for list-semantics.

Reply:
+1 list<> to vector<>